diff -Naur libiconv-1.14/build-aux/config.sub libiconv-1.14/build-aux/config.sub
--- libiconv-1.14/build-aux/config.sub	2009-06-21 16:47:33.000000000 +0530
+++ libiconv-1.14/build-aux/config.sub	2016-02-26 23:12:48.000000000 +0530
@@ -322,6 +322,7 @@
 	| alpha64-* | alpha64ev[4-8]-* | alpha64ev56-* | alpha64ev6[78]-* \
 	| alphapca5[67]-* | alpha64pca5[67]-* | arc-* \
 	| arm-*  | armbe-* | armle-* | armeb-* | armv*-* \
+	| aarch64-* \
 	| avr-* | avr32-* \
 	| bfin-* | bs2000-* \
 	| c[123]* | c30-* | [cjt]90-* | c4x-* | c54x-* | c55x-* | c6x-* \
diff -Naur libiconv-1.14/lib/Makefile.in libiconv-1.14/lib/Makefile.in
--- libiconv-1.14/lib/Makefile.in	2011-08-07 23:18:03.000000000 +0530
+++ libiconv-1.14/lib/Makefile.in	2016-04-24 18:38:22.000000000 +0530
@@ -68,7 +68,7 @@
 all : libiconv.la
 
 libiconv.la : $(OBJECTS)
-	$(LIBTOOL_LINK) $(CC) $(LDFLAGS) $(CFLAGS) -o libiconv.la -rpath $(libdir) -version-info $(LIBICONV_VERSION_INFO) -no-undefined $(OBJECTS)
+	$(LIBTOOL_LINK) $(CC) $(LDFLAGS) $(CFLAGS) -o libiconv.la -rpath $(libdir) -avoid-version -no-undefined $(OBJECTS)
 
 iconv.lo : $(srcdir)/iconv.c $(srcdir)/converters.h \
            $(srcdir)/encodings.def $(srcdir)/encodings_aix.def $(srcdir)/encodings_osf1.def $(srcdir)/encodings_dos.def $(srcdir)/encodings_local.def \
diff -Naur libiconv-1.14/libcharset/build-aux/config.sub libiconv-1.14/libcharset/build-aux/config.sub
--- libiconv-1.14/libcharset/build-aux/config.sub	2009-06-21 16:47:33.000000000 +0530
+++ libiconv-1.14/libcharset/build-aux/config.sub	2016-02-26 23:14:17.000000000 +0530
@@ -317,6 +317,7 @@
 	| alpha64-* | alpha64ev[4-8]-* | alpha64ev56-* | alpha64ev6[78]-* \
 	| alphapca5[67]-* | alpha64pca5[67]-* | arc-* \
 	| arm-*  | armbe-* | armle-* | armeb-* | armv*-* \
+	| aarch64-* \
 	| avr-* \
 	| bfin-* | bs2000-* \
 	| c[123]* | c30-* | [cjt]90-* | c4x-* | c54x-* | c55x-* | c6x-* \
diff -Naur libiconv-1.14/libcharset/lib/Makefile.in libiconv-1.14/libcharset/lib/Makefile.in
--- libiconv-1.14/libcharset/lib/Makefile.in	2009-06-21 16:47:33.000000000 +0530
+++ libiconv-1.14/libcharset/lib/Makefile.in	2016-04-24 18:38:43.000000000 +0530
@@ -56,7 +56,7 @@
 all : libcharset.la charset.alias ref-add.sed ref-del.sed
 
 libcharset.la : $(OBJECTS)
-	$(LIBTOOL_LINK) $(CC) $(LDFLAGS) $(CFLAGS) -o libcharset.la -rpath $(libdir) -version-info $(LIBCHARSET_VERSION_INFO) -no-undefined $(OBJECTS)
+	$(LIBTOOL_LINK) $(CC) $(LDFLAGS) $(CFLAGS) -o libcharset.la -rpath $(libdir) -avoid-version -no-undefined $(OBJECTS)
 
 localcharset.lo : $(srcdir)/localcharset.c
 	$(LIBTOOL_COMPILE) $(CC) $(INCLUDES) $(CFLAGS) $(CPPFLAGS) $(DEFS) -c $(srcdir)/localcharset.c
