diff -Naur libsecret-0.18.4/tool/secret-tool.c libsecret-0.18.4/tool/secret-tool.c
--- libsecret-0.18.4/tool/secret-tool.c	2014-09-23 13:19:21.000000000 +0530
+++ libsecret-0.18.4/tool/secret-tool.c	2016-02-05 08:14:44.000000000 +0530
@@ -33,6 +33,40 @@
 static gchar *store_label = NULL;
 static gchar *store_collection = NULL;
 
+#ifdef __ANDROID__
+#include <stdio.h>
+#include <termios.h>
+static gchar *getpass() {
+    struct termios oflags, nflags;
+    char password[64];
+    gchar *pw;
+
+    /* disabling echo */
+    tcgetattr(fileno(stdin), &oflags);
+    nflags = oflags;
+    nflags.c_lflag &= ~ECHO;
+    nflags.c_lflag |= ECHONL;
+
+    if (tcsetattr(fileno(stdin), TCSANOW, &nflags) != 0) {
+        perror("tcsetattr");
+        return NULL;
+    }
+
+    fgets(password, sizeof(password), stdin);
+    password[strlen(password) - 1] = 0;
+
+    /* restore terminal */
+    if (tcsetattr(fileno(stdin), TCSANOW, &oflags) != 0) {
+        perror("tcsetattr");
+        return NULL;
+    }
+    
+	pw = (gchar *)malloc(strlen(password+1));
+	strcpy(pw, password);
+	return pw;    
+}
+#endif
+
 /* secret-tool store --label="blah" --collection="xxxx" name:xxxx name:yyyy */
 static const GOptionEntry STORE_OPTIONS[] = {
 	{ "label", 'l', 0, G_OPTION_ARG_STRING, &store_label,
